components/esp-tls/esp_tls_mbedtls.c

```diff
@@ -1073,6 +1075,26 @@ static esp_err_t esp_set_atecc608a_pki_context(esp_tls_t *tls, const void *pki)
     }
     mbedtls_x509_crt_init(&tls->clientcert);
 
+    esp_tls_pki_t *pki_l = (esp_tls_pki_t *) pki;
+    if (pki_l->publiccert_pem_buf != NULL) {
+        ret = mbedtls_x509_crt_parse(&tls->clientcert, pki_l->publiccert_pem_buf, pki_l->publiccert_pem_bytes);
+        if (ret < 0) {
+            ESP_LOGE(TAG, "mbedtls_x509_crt_parse of client cert returned -0x%04X", -ret);
+            mbedtls_print_error_msg(ret);
+            ESP_INT_EVENT_TRACKER_CAPTURE(tls->error_handle, ESP_TLS_ERR_TYPE_MBEDTLS, -ret);
+            return ESP_ERR_MBEDTLS_X509_CRT_PARSE_FAILED;
+        }
+    } else {
+        ESP_LOGE(TAG, "Device certificate must be provided for TrustCustom Certs");
+        return ESP_FAIL;
+    }
```

Comment:
Detta är kod som aktiveras för `CONFIG_ATECC608A_TCUSTOM`, är det avsiktligt?

Reply:
Det ser konstigt ut, men om du tänker efter så är ju koden från rad 1098 och nedanför oförändrad, det är den tidigare TCUSOM-koden. Så TCUSTOM och MANUAL skall vara samma, förutom att vi i MANUAL inte activerar atecc. MEN nu verkar det som jag ändå aktiverar atecc-chippet så jag borde nog helt enkelt köra samma kod för MANUAL och TCUSTOM. Vet inte om MANUAL ens behövs...